feat(docs-infra): support contributors belonging to multiple groups (and other improvements)

gkalpak opened the pull request:
Several improvements related to the contributors page: 
- docs: change contributor group name (Collaborator --> Collaborators)
- docs: change heading of contributors page to avoid confusion (Collaborators --> Contributors)
- docs: change the order of groups in contributors page
- refactor(docs-infra): change unused classes to interfaces
- feat(docs-infra): support contributors belonging to multiple groups
- docs: add Sam Julien to GDE group as well
- build(docs-infra): add check to ensure all contributor pictures exist

(See individual commits for details.)

mary-poppins commented:
You can preview f0a5970 at https://pr29553-f0a5970.ngbuilds.io/.

petebacondarwin commented on aio/content/marketing/contributors.json:
```diff
@@ -658,7 +658,7 @@
     "website": "http://www.samjulien.com/",
     "bio": "Sam Julien builds software, articles, video courses, and campfires. A developer, speaker, writer, and GDE in the Pacific Northwest, Sam's favorite thing in the world is changing someone's life by teaching them to code.",
-    "groups": ["Collaborators"],
+    "groups": ["Collaborators", "GDE"],
```
All this work for @samjulien - I hope you are happy now!

gkalpak replied:
:joy: Yeah, I didn't realize it was only Sam until after implementing it.
I am pretty sure there are more to come (I know of at least one more 😉).

jasonaden commented:
This doesn't merge cleanly with `patch`. Looks like there's at least a file missing on `patch` branch, and maybe some other conflicts. It should ideally be split into two PRs.

petebacondarwin commented:
I think it doesn't merge cleanly because the original PR that added Contributors did not land on patch?

gkalpak commented:
Yes, the original PR was only merged into  master: #28930 
Then, it was backborted as 34395aeea, but some changes were missed.

Backporting...

gkalpak commented:
Created #29584. Once that lands, this PR should merge cleanly into both master and patch.
Marking this as blocked on #29584.

gkalpak committed:
docs: change the order of groups in contributors page

Put "Collaborators" before "GDE", since they are semantically closer to
the core team.

gkalpak committed:
docs: change contributor group name (Collaborator --> Collaborators)

The name of the group also serves as the group tab's title in the docs
app and having "Collaborator" (singular) as a title for a list of people
didn't read well.
Changing th group name (and thus tab title) to "Collaborators" (plural).

gkalpak committed:
docs: change heading of contributors page to avoid confusion (Collaborators --> Contributors)

Now that "Collaborators" is a separate group in the contributors page,
having "Angular Collaborators" as the heading of the page (which also
contains the "Angular" and "GDE" groups) is confusing.
Changing the title to "Angular Contributors" to avoid confusion.

gkalpak committed:
refactor(docs-infra): change unused classes to interfaces

gkalpak committed:
docs: add Sam Julien to GDE group as well

gkalpak committed:
feat(docs-infra): support contributors belonging to multiple groups

gkalpak committed:
build(docs-infra): add check to ensure all contributor pictures exist

gkalpak commented:
In #29584, I noticed that one contributor picture was missing, so it appeared blank on the page. So, I also added one last commit to ensure that all contributor pictures (referenced in `contributors.json`) do actually exist.

@petebacondarwin, PTAL.

mary-poppins commented:
You can preview eef6512 at https://pr29553-eef6512.ngbuilds.io/.

petebacondarwin commented:
Not looking so good @gkalpak  https://pr29553-eef6512.ngbuilds.io/about

gkalpak committed:
fixup! feat(docs-infra): support contributors belonging to multiple groups

mary-poppins commented:
You can preview 43c038e at https://pr29553-43c038e.ngbuilds.io/.

petebacondarwin commented:
LOL https://pr29553-43c038e.ngbuilds.io/about?group=Collaborator

gkalpak committed:
fixup! feat(docs-infra): support contributors belonging to multiple groups

gkalpak commented:
🤦‍♂️

mary-poppins commented:
You can preview eb77827 at https://pr29553-eb77827.ngbuilds.io/.
